crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -751,14 +733,8 @@ impl<S: DatabaseExt + Send, H: HistoryMode> VmTracer<EraDb<S>, H> for CheatcodeT
                 }
                 FinishCycleOneTimeActions::RevertToSnapshot { snapshot_id } => {
                     let mut storage = storage.borrow_mut();
-
-                    let modified_storage: HashMap<StorageKey, H256> = storage
-                        .modified_storage_keys()
-                        .clone()
-                        .into_iter()
-                        .filter(|(key, _)| key.address() != &zksync_types::SYSTEM_CONTEXT_ADDRESS)
-                        .collect();
-
+                    let modified_storage =
+                        self.get_modified_storage(storage.modified_storage_keys());
```

Comment:
before you used only the latest modified storages, but now you are using it with a combination of previously applied storages. is it correct?